simulator/examples/polyline-joints.rs

```diff
@@ -0,0 +1,97 @@
+use embedded_graphics::{
+    pixelcolor::Rgb888, prelude::*, primitives::Line, primitives::*, style::PrimitiveStyleBuilder,
+};
+use embedded_graphics_simulator::{
+    OutputSettingsBuilder, SimulatorDisplay, SimulatorEvent, Window,
+};
+use sdl2::keyboard::Keycode;
+
+fn draw(
+    points: &[Point],
+    width: u32,
+    mouse_pos: Point,
+    display: &mut SimulatorDisplay<Rgb888>,
+) -> Result<(), core::convert::Infallible> {
+    display.clear(Rgb888::BLACK)?;
+
+    Polyline::new(points)
+        .into_styled(
+            PrimitiveStyleBuilder::new()
+                .stroke_width(width)
+                .stroke_color(Rgb888::RED)
+                .build(),
+        )
+        .draw(display)
+}
+
+const PADDING: i32 = 16;
+
+fn main() -> Result<(), core::convert::Infallible> {
+    let w = 320i32;
+    let h = 180i32;
+
+    let mut display: SimulatorDisplay<Rgb888> =
+        SimulatorDisplay::new(Size::new(w as u32 + 100, h as u32));
+    let output_settings = OutputSettingsBuilder::new().scale(4).build();
+    let mut window = Window::new("Polyline joints debugger", &output_settings);
+
+    let mut end_point = Point::new(363, 39);
+
+    let mut width = 15u32;
+
+    let mut mouse_down = false;
+
+    let mut points = [
+        Point::new(PADDING, h / 2),
+        Point::new(100, h / 2),
+        Point::new(120, h / 2 - 20),
+        Point::new(140, h / 2),
+        Point::new(160, h / 2),
+        Point::new(180, h / 2 + 10),
+        Point::new(200, PADDING),
+        Point::new(220, h / 2 + 20),
+        Point::new(240, h / 2),
+        Point::new(w - PADDING, h / 2),
+        end_point,
+    ];
+
+    let mut num_points = points.len();
+
+    draw(&points, width, end_point, &mut display)?;
+
+    'running: loop {
+        window.update(&display);
+
+        for event in window.events() {
+            match event {
+                SimulatorEvent::Quit => break 'running,
+                SimulatorEvent::MouseButtonDown { point, .. } => {
+                    mouse_down = true;
+
+                    end_point = point;
+                }
+                SimulatorEvent::KeyDown { keycode, .. } => match keycode {
```

Comment:
There seems to be a problem with determining when a bevel join should be used. For some small stroke widths the top vertex is drawn using a bevel join:
![grafik](https://user-images.githubusercontent.com/226123/97708895-c4535680-1ab9-11eb-86d7-9c901f976721.png)

But if the stroke width is increased by one 1px the miter reappears:
![grafik](https://user-images.githubusercontent.com/226123/97708938-d46b3600-1ab9-11eb-9fd6-261578a22754.png)

This pattern continues for the next few stroke widths until the vertex stays beveled.

Reply:
I've seen this as well. I think it's an unlucky combination of the joint angle in the demo and its effect on the bevel heuristic. Tbh I'm ok leaving it as is, but do you have a suggestion on how to fix it better?

Reply:
I don't think we need to fix this in this PR, but IMO we should take another look at this later. Can you open an issue for this?